dietpi/dietpi-config

```diff
@@ -430,33 +430,95 @@ A long (or insufficiently manufactured) cable may required a higher boost settin
 
 		elif (( $G_WHIP_RETURNED_VALUE == 16 )); then
 
-			# Known FP's for setting brightness
-			local afp_current_set_brightness=(
-
-				'/sys/class/backlight/intel_backlight'
-				'/sys/class/backlight/rpi_backlight'
-				'/sys/class/backlight/lcd0'
-				'/sys/class/backlight/backlight'
-			)
-
-			MIN_VALUE=0
-			MAX_VALUE=255
-			local fp_brightness current_brightness
-			for fp_brightness in "${afp_current_set_brightness[@]}"
+			local aFp_Brightness_Available=()
+			local aBrightness_Name=()
+			local aBrightness_Current=()
+			local aBrightness_Max=()
+
+			local iTotalAvailable=0
+			local fp_brightness
+			for fp_brightness in /sys/class/backlight/*
 			do
-				[[ -f $fp_brightness/brightness ]] || continue
-				current_brightness=$(<"$fp_brightness/brightness")
-				[[ -f $fp_brightness/max_brightness ]] && MAX_VALUE=$(<"$fp_brightness/max_brightness")
-				break
+				if [[ -f $fp_brightness/brightness && -f $fp_brightness/max_brightness && -f $fp_brightness/actual_brightness ]] &&
+					disable_error=1 G_CHECK_VALIDINT "$(<"$fp_brightness/brightness")" &&
+					disable_error=1 G_CHECK_VALIDINT "$(<"$fp_brightness/max_brightness")" &&
+					disable_error=1 G_CHECK_VALIDINT "$(<"$fp_brightness/actual_brightness")"
+				then
+					aFp_Brightness_Available+=( "$fp_brightness" )
+					aBrightness_Name+=( "${fp_brightness##*/}" )
+					aBrightness_Current+=( "$(<"$fp_brightness/actual_brightness")" )
+					aBrightness_Max+=( "$(<"$fp_brightness/max_brightness")" )
+					(( iTotalAvailable++ ))
+				fi
```

Comment:
I think I'd take the risk and for now trust in that if a device is registered as backlight device, the sysfs API is complete. It doesn't seem to be an API which changed recently:
```suggestion
				aFp_Brightness_Available+=( "$fp_brightness" )
				aBrightness_Name+=( "${fp_brightness##*/}" )
				aBrightness_Current+=( "$(<"$fp_brightness/actual_brightness")" )
				aBrightness_Max+=( "$(<"$fp_brightness/max_brightness")" )
				(( iTotalAvailable++ ))
```